### ISSUE TITLE
ec pairing: panic inside final_exp

### ISSUE THREAD

**Issue by lispc** [scroll-tech/zkevm-circuits#876]
### Describe the bug

cmd
```
C="ecpairing_two_point_oog_d0_g0_v0"
FLAG=" --features=scroll"
export CIRCUIT=ecc
RUST_BACKTRACE=1 RUST_LOG=trace cargo run --release $FLAG -- --circuits basic --suite nightly --inspect "${C}" 
```

output:
```
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `0`,
 right: `1`', /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/subtle-2.4.1/src/lib.rs:612:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/dfe3fe710181738a2cb3060c23ec5efb3c68ca09/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/dfe3fe710181738a2cb3060c23ec5efb3c68ca09/library/core/src/panicking.rs:64:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
   4: halo2_ecc::fields::FieldChip::divide_unsafe
   5: halo2_ecc::bn254::final_exp::<impl halo2_ecc::fields::fp12::Fp12Chip<F,halo2_ecc::fields::fp::FpConfig<F,halo2curves::bn256::fq::Fq>,halo2curves::bn256::fq12::Fq12,9_i64>>::cyclotomic_decompress
   6: halo2_ecc::bn254::final_exp::<impl halo2_ecc::fields::fp12::Fp12Chip<F,halo2_ecc::fields::fp::FpConfig<F,halo2curves::bn256::fq::Fq>,halo2curves::bn256::fq12::Fq12,9_i64>>::cyclotomic_pow
   7: halo2_ecc::bn254::final_exp::<impl halo2_ecc::fields::fp12::Fp12Chip<F,halo2_ecc::fields::fp::FpConfig<F,halo2curves::bn256::fq::Fq>,halo2curves::bn256::fq12::Fq12,9_i64>>::final_exp
```

**Comment by lispc** [comment]
5 panics in total

```
[2023-09-02T04:35:40Z INFO  testool::statetest::results] 💀PANIC ecpairing_two_point_oog_d0_g0_v0 assertion failed: `(left == right)`
[2023-09-02T04:38:58Z INFO  testool::statetest::results] 💀PANIC ecpairing_two_point_oog_d0_g3_v0 assertion failed: `(left == right)`
[2023-09-02T06:11:03Z INFO  testool::statetest::results] 💀PANIC ecpairing_two_point_match_2_d0_g0_v0 assertion failed: `(left == right)`
[2023-09-02T06:14:27Z INFO  testool::statetest::results] 💀PANIC ecpairing_two_point_match_2_d0_g3_v0 assertion failed: `(left == right)`
[2023-09-02T06:17:48Z INFO  testool::statetest::results] 💀PANIC ecpairing_two_point_match_1_d0_g0_v0 assertion failed: `(left == right)`
```

**Comment by roynalnaruto** [comment]
Closed by the updated made in #889
